src/pkg/build/conveyorPacker_busybox.go

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package build
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/singularityware/singularity/src/pkg/sylog"
+)
+
+// BusyBoxConveyor only needs to hold the conveyor to have the needed data to pack
+type BusyBoxConveyor struct {
+	recipe Definition
+	src    string
+	tmpfs  string
+	b      *Bundle
+}
+
+// BusyBoxConveyorPacker only needs to hold the conveyor to have the needed data to pack
+type BusyBoxConveyorPacker struct {
+	BusyBoxConveyor
+}
+
+// Get just stores the source
+func (c *BusyBoxConveyor) Get(recipe Definition) (err error) {
+
+	c.recipe = recipe
+
+	c.b, err = NewBundle("")
+	if err != nil {
+		return
+	}
+
+	//get mirrorURL, OSVerison, and Includes components to definition
+	mirrorurl, ok := recipe.Header["mirrorurl"]
+	if !ok {
+		return fmt.Errorf("Invalid busybox header, no MirrurURL specified")
+	}
+
+	err = c.insertBaseEnv()
+	if err != nil {
+		return fmt.Errorf("While inserting base environment: %v", err)
+	}
+
+	err = c.insertBaseFiles()
+	if err != nil {
+		return fmt.Errorf("While inserting files: %v", err)
+	}
+
+	busyBoxPath, err := c.insertBusyBox(mirrorurl)
+	if err != nil {
+		return fmt.Errorf("While inserting busybox: %v", err)
+	}
+
+	cmd := exec.Command(busyBoxPath, `--install`, filepath.Join(c.b.Rootfs(), "/bin"))
+
+	sylog.Debugf("\n\tBusyBox Path: %s\n\tMirrorURL: %s\n", busyBoxPath, mirrorurl)
+
+	err = cmd.Run()
+	if err != nil {
+		return fmt.Errorf("While performing busybox install: %v", err)
+	}
+
+	return nil
+}
+
+// Pack puts relevant objects in a Bundle!
+func (cp *BusyBoxConveyorPacker) Pack() (b *Bundle, err error) {
+
+	err = cp.insertRunScript()
+	if err != nil {
+		return nil, fmt.Errorf("While inserting base environment: %v", err)
+	}
+
+	cp.b.Recipe = cp.recipe
+
+	return cp.b, nil
+}
+
+func (c *BusyBoxConveyor) insertBaseFiles() (err error) {
+
+	fp, err := os.Create(filepath.Join(c.b.Rootfs(), "/etc/passwd"))
```

Comment:
maybe `ioutil.Writefile` offers a less verbose pattern... you can create the file, add the content, and set the mode in one step - `https://golang.org/pkg/io/ioutil/#WriteFile`

Reply:
Agreed. Much cleaner.